src/medCore/gui/toolboxes/medSelectorToolBox.cpp

```diff
@@ -0,0 +1,173 @@
+/*=========================================================================
+
+ medInria
+
+ Copyright (c) INRIA 2013 - 2014. All rights reserved.
+ See LICENSE.txt for details.
+ 
+  This software is distributed WITHOUT ANY WARRANTY; without even
+  the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+  PURPOSE.
+
+=========================================================================*/
+
+#include <medAbstractToolBox.h>
+#include <medSelectorToolBox.h>
+#include <medToolBoxFactory.h>
+#include <medToolBoxHeader.h>
+#include <medToolBoxTab.h>
+
+class medSelectorToolBoxPrivate
+{
+public:
+    medComboBox *chooseComboBox;
+    medAbstractToolBox * currentToolBox;
+    QHash<QString, medAbstractToolBox*> toolBoxes;
+    QVBoxLayout *mainLayout;
+
+    medAbstractData *inputData;
+};
+
+medSelectorToolBox::medSelectorToolBox(QWidget *parent, QString tlbxId) :
+    medToolBox(parent),
+    d(new medSelectorToolBoxPrivate)
+{
+    d->currentToolBox = NULL;
+
+    d->chooseComboBox = new medComboBox;
+    d->chooseComboBox->addItem("Choose algorithm");
+    d->chooseComboBox->setToolTip(tr("Browse through the list of available algorithm"));
```

Comment:
Although, this two texts aren't applicable to all workspaces now. Maybe replace by "choose toolbox" for example.